File: python/array/twoSum.py
from typing import List
class Solution:
    def two_sum(self, nums:List[int], target:int)->List[int]:

        #dictionarry to store numbers and their indices
        hashmap = {}

        #loop through the array
        for i, num in enumerate(nums):
            #get diffreence
            diff = target - num

            #check if the diffrence is in hashmap
            if diff in hashmap:
               return[hashmap[diff], i]

            #add to hashmap if not found
            hashmap[num] = i
        return []
    
if __name__ == "__main__":
    solution = Solution()
    print(solution.two_sum([2,7,11,15],18))

File: java/array/twoSum.java
import java.util.HashMap;
import java.util.Map;

class Solution {
    public int[] twoSum(int[] nums, int target) {

        // HashMap to store numbers and their indices.
        Map<Integer, Integer> map = new HashMap<>();

        // Iterate through the array.
        for (int i = 0; i < nums.length; i++) {

            // Difference needed to reach target
            int diff = target - nums[i];

            // Check if difference exists in map
            if (map.containsKey(diff)) {
                return new int[]{map.get(diff), i};
            }

            // Store current number and its index
            map.put(nums[i], i);
        }

        // No solution found 
        return new int[]{};
    }
}